apps/tutorial_servo_control_can/main.cpp

```diff
@@ -0,0 +1,43 @@
+#include "TutorialServo.h"
+#include "CAN.h"
+#include "CANMsg.h"
+#include "hw_bridge.h"
+#include <cstdio>
+
+
+TutorialServo servo_can(PA_1); //create servo
+
+//initialize CAN
+
+CANMsg rxMsg;
+CANMsg txMsg;
+#define TX_ID 0x100
+#define RX_ID 0x101
+CAN can(CAN_RX, CAN_TX, HWBRIDGE::ROVERCONFIG::ROVER_CANBUS_FREQUENCY);
+
+int main(){
+    float percent; 
+    rxMsg.id = RX_ID; //set message ID
+
+    while(1){
+
+        if (can.read(rxMsg)) {
+  
+        printf("-------------------------------------\r\n");
+        printf("CAN message received\r\n");
+    
+
+    // Filtering performed by software:
+        if (rxMsg.id == RX_ID) {
+            rxMsg >> percent;  // extract data from the received CAN message
+        
+                servo_can.setServoPositionWithPercent(percent);

```

Comment:
the challenge asks u to use the setpositionwithdegrees function

Reply:
"The program should be able to receive CAN messages which contain a percentage of the range of motion of the servo as a float." The training said to take in a percent ?

Reply:
ur right.